src/metpy/plots/declarative.py

```diff
@@ -1831,3 +1835,266 @@ def _build(self):
+    A single string (color name or hex code) or collection of strings for the outline color of
+    labels. If a collection, the first color corresponds to the label of the first Shapely
+    object in `geometry`, the second color corresponds to the label of the second Shapely
+    object, and so on. Default value is `fill`.
+    """
+
+    @staticmethod
+    @validate('geometry')
+    def _valid_geometry(_, proposal):
+        """Cast `geometry` into a list once it is provided by user.
+
+        Users can provide any kind of collection, such as a ``GeoPandas.GeoSeries``, and this
+        turns them into a list.
+        """
+        geometry = proposal['value']
+        return list(geometry)
+
+    @staticmethod
+    @validate('fill', 'stroke', 'label_facecolor', 'label_edgecolor')
+    def _valid_color_list(_, proposal):
+        """Cast color-related attributes into a list once provided by user.
+
+        This is necessary because _build() expects to cycle through a list of colors when
+        assigning them to the geometry.
+        """
+        color = proposal['value']
+
+        if isinstance(color, str):
+            color = [color]
+        # `color` must be a collection if it is not a string
+        else:
+            color = list(color)
+
+        return color
+
+    @staticmethod
+    @validate('labels')
+    def _valid_labels(_, proposal):
+        """Cast `labels` into a list, if not None, once provided by user."""
+        labels = proposal['value']
+        labels = list(labels) if labels is not None else None
+        return labels
+
+    @observe('fill', 'stroke')
+    def _update_label_colors(self, change):
+        """Set default text colors using `fill` and `stroke`.
+
+        If `label_facecolor` or `label_edgecolor` have not been specified, provide default
+        colors for those attributes using `fill` and `stroke`.
+        """
+        if change['name'] == 'fill' and self.label_edgecolor is None:
+            self.label_edgecolor = self.fill
+        elif change['name'] == 'stroke' and self.label_facecolor is None:
+            self.label_facecolor = self.stroke
+
+    @property
+    def name(self):
+        """Generate a name for the plot."""
+        # Unlike Plots2D and PlotObs, there are no other attributes (such as 'fields' or
+        # 'levels') from which to name the plot. A generic name is returned here in case the
+        # user does not provide their own title, in which case MapPanel.draw() looks here.
+        return 'Geometry Plot'
+
+    @staticmethod
+    def _position_label(geo_obj, label):
+        """Return a (lon, lat) where the label of a polygon/line/point can be placed."""
+        # We hash the label, which is used in choosing a point along the polygon or line that
+        # will be returned. This "psuedo-randomizes" the position of a label, in hopes of
+        # spatially dispersing the labels and lessening the chance that labels overlap.
+        label_hash = sum(map(ord, str(label)))
```

Comment:
Any chance you could use the builtin `hash` instead?

Reply:
The reason I shied away from `hash` was that Python uses a random seed to come up with the hash for an object. So every time a user plots labels with `PlotGeometry` in a new kernel/session, the hash will be different and their labels will end up in different locations on the plot (for no apparent reason from the user's perspective). With the current hash, labels will always plot in the same place every time the plot is drawn.

Reply:
Great point.